ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/BlockTree.hs

```diff
@@ -186,22 +185,28 @@ prettyPrint blockTree = do
       & ("─" ++)
       & (printf "%2d" (unBlockNo oBlockNo) ++)
       & ("  trunk: " ++)
-    ,
-
-    advFragment
-      & AF.toOldestFirst
-      & map (\block -> (fromIntegral (unSlotNo (blockSlot block) - unSlotNo aoSlotNo - 1), Just (unBlockNo (blockNo block))))
-      & Vector.toList . (Vector.replicate (fromIntegral (unSlotNo ahSlotNo - unSlotNo aoSlotNo)) Nothing Vector.//)
-      & map (maybe "  " (printf "%2d"))
-      & unwords
-      & map (\c -> if c == ' ' then '─' else c)
-      & (" ╰─" ++)
-      & (replicate (3 * fromIntegral (unSlotNo (aoSlotNo - oSlotNo))) ' ' ++)
-      & ("         " ++)
-    ]
+    ] ++ advBranches
 
   where
     slotAndBlockNoFromAnchor :: AF.Anchor b -> (SlotNo, BlockNo)
     slotAndBlockNoFromAnchor = \case
       AF.AnchorGenesis -> (0, 0)
       AF.Anchor slotNo _ blockNo' -> (slotNo, blockNo')
+
+    branch :: SlotNo -> AF.AnchoredFragment blk -> (String, SlotNo, SlotNo)
+    branch oSlotNo frag =
+      (blocks, aoSlotNo, ahSlotNo)
+      where
+        blocks =
+          frag
+            & AF.toOldestFirst
+            & map (\block -> (fromIntegral (unSlotNo (blockSlot block) - unSlotNo aoSlotNo - 1), Just (unBlockNo (blockNo block))))
+            & Vector.toList . (Vector.replicate (fromIntegral (unSlotNo ahSlotNo - unSlotNo aoSlotNo)) Nothing Vector.//)
```

Comment:
Frankly, I'm not even exactly sure what this is doing; it would be worth documenting, or even rewriting completely.

Reply:
😳 wtf 😅